desc/objectives/_coils.py

```diff
@@ -0,0 +1,504 @@
+import numbers
+
+import jax
```

Comment:
we try not to import jax directly, in order to keep an interface layer in case they change things.

If you need specific functions (like the tree map stuff) import it into `desc.backend` and then here do `from desc.backend import tree_map`